crates/dojo/core/src/meta/introspect.cairo

```diff
@@ -1,5 +1,164 @@
 use dojo::meta::Layout;
 use dojo::storage::packing;
+use core::panics::panic_with_byte_array;
+
+// Each index matches with a primitive types in both arrays (main and nested).
+// The main array represents the source primitive while nested arrays represents
+// destination primitives.
+// 'bool': 0
+// 'u8': 1
+// 'u16': 2
+// 'u32' / 'usize': 3
```

Comment:
Even if for now `usize` is 32 bits, I'm wondering if this type shouldn't be a max value.
Since `usize` is supposed to be portable, we don't want one to upgrade a `usize` since it's supposed to represent the biggest value supported by the underlying memory architecture.

So `usize` shouldn't be changed, wdyt? If you agree, @Larkooo the same behavior will be replicated on Torii.

Reply:
> Even if for now `usize` is 32 bits, I'm wondering if this type shouldn't be a max value. Since `usize` is supposed to be portable, we don't want one to upgrade a `usize` since it's supposed to represent the biggest value supported by the underlying memory architecture.
> 
> So `usize` shouldn't be changed, wdyt? If you agree, @Larkooo the same behavior will be replicated on Torii.

Hum yes, I agree. I wonder if we shouldn't ban the usage of `usize` in `models`/`events` as the felt representation of it is not really fixed. At compile time, we could output a message saying to use a known size type as `u32` instead of an architecture-dependent size type such as `usize`.

WDYT @glihm and @Larkooo ?

Reply:
I think it makes sense yes, otherwise we'll end up having different behaviour across toriis, regardless of being able to upgrade it

Reply:
Yep let's go this way @remybar. 👍